tests/test_abacus_md.py

```diff
@@ -40,6 +50,18 @@ def test_coord(self) :
                     self.assertAlmostEqual(self.system_water.data['coords'][ii][jj][kk], coord[ii][jj][kk])
         fp.close()
 
+        fp2 = open('abacus.md.nostress/Si_coord')
```

Comment:
`with open("abacus.md.nostress/Si_coord") as fp2:` is better. When the assertion `self.assertAlmostEqual` fails,  in your implementation `fp2.close()` will not be exceuted.

Reply:
I have understood the problem. Now the `fp = open('...')` codes in the `test_abacus_md.py` have all been replaced with `with open('...') as fp:`. I will fix the cases in `test_abacus_scf.py` in the next pr.